src/components/DistanceRequest.js

```diff
@@ -161,6 +161,13 @@ function DistanceRequest({iou, iouType, report, transaction, mapboxAccessToken})
         Transaction.getRoute(iou.transactionID, validatedWaypoints);
     }, [shouldFetchRoute, iou.transactionID, validatedWaypoints, isOffline]);
 
+    useEffect(() => {
+        if (numberOfWaypoints <= numberOfPreviousWaypoints) {
```

Comment:
Why change this to check for a decrease in the number of waypoints rather than an increase?